### DISCUSSION TITLE
Squash commits along with their history

### DISCUSSION THREAD

**Discussion by hellridergit** [community/community#73257] [Repositories] [answered]
At some point I decided some commits I made in the main branch were unnecessary and redundant, so I 'rebase squashed' them, and they correctly disappeared from the branch.

However, those squashed commits were still visible (with details) on the repository activity page, and I would like to make them disappear from there as well.

### Question 
How can I achieve that without having to wait for the GitHub's garbage collection cycle?

I don't need a retroactive fix. It'd be sufficient to know how to do it in the future.

Thanks.

**Comment by GigaSmurf** [comment, marked as answer]
On GitHub, once commits are pushed to the repository, they become a part of the repository's history and are visible in the various activity logs and timelines, even if you later rebase and squash them.

If you want to ensure that those commits are not visible on GitHub in the future, you'll need to avoid pushing them to the remote repository. If you've already pushed them, here's what you could do:
1. Create a Fresh clone: clone the repository to a new location, perform the rebase and squash in this fresh clone, and force push the newly rebased branch to overwrite the remote branch.
2. If there's sensitive data, you can contact Github Support.

For future operations, try using a development branch, squashing before pushing, amending commits locally, or just rebasing locally.

**Reply by hellridergit** [reply]
Thank you, will explore your suggestions when needed!
